src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp

```diff
@@ -2066,14 +2066,20 @@ void MacroAssembler::membar(Membar_mask_bits order_constraint) {
   address last = code()->last_insn();
   if (last != nullptr && nativeInstruction_at(last)->is_Membar() && prev == last) {
     NativeMembar *bar = NativeMembar_at(prev);
-    // We are merging two memory barrier instructions.  On AArch64 we
-    // can do this simply by ORing them together.
-    bar->set_kind(bar->get_kind() | order_constraint);
-    BLOCK_COMMENT("merged membar");
-  } else {
-    code()->set_last_insn(pc());
-    dmb(Assembler::barrier(order_constraint));
+    // We need avoid promoting barrier to dmb.ish,
```

Comment:
```suggestion
    // Don't promote DMB ST|DMB LD to DMB (a full barrier) because
    // doing so would introduce  a StoreLoad which  the caller did not
    // intend.
```
I think that should be clear enough.

Reply:
Thanks for comments. Fixed.